### DISCUSSION TITLE
How to use an external lib that uses StyleX?

### DISCUSSION THREAD

**Discussion by olivierpascal** [facebook/stylex#343] [Q&A] [answered]
I have a Next.js project which depends of a lib that uses StyleX. How to make it works without including the whole generated `stylex.css` of the lib?

**Comment by olivierpascal** [comment]
Why I don't want to just import the compiled `stylex.css`:
- I want styles to be compiled at runtime, only for the actually used component from the third-party lib ;
- I want to share css vars ;

Is this usecase covered by stylex? Do I have to use monorepo for that?

**Comment by olivierpascal** [comment]
So I guess the solution for that is monorepo. I am digging. Related topics:
- https://github.com/facebook/stylex/issues/63
- https://github.com/facebook/stylex/discussions/204
- https://github.com/facebook/stylex/discussions/276
- https://github.com/facebook/stylex/discussions/111

**Comment by necolas** [comment, marked as answer]
The lib shouldn't be compiling styles before publish. When you bundle the components in your project and any components from dependencies using StyleX, that is when StyleX  should be used to compile all styles and a bundler can extract them to a CSS file

**Reply by olivierpascal** [reply]
Mm yes that's what I figured out but it still does not work unfortunately as per my last edit.

**Reply by necolas** [reply]
Is your bundler ignoring directories like node_modules? That would prevent babel from processing files

**Reply by olivierpascal** [reply]
I don't ask him specifically to do so. I am using nextjs.

**Reply by olivierpascal** [reply]
Sorry, I was on my phone. When I wrote about 'my last edit', I was referencing this discussion https://github.com/facebook/stylex/discussions/228#discussioncomment-8110290

So maybe this is a nextjs specific issue.

**Reply by nmn** [reply]
NextJS ignores `node_modules` by default. You can use the `transpilePackages` option to make it compile certain packages. See the NextJS example in the repo and how it include `@stylexjs/open-props` as an example.

https://nextjs.org/docs/app/api-reference/next-config-js/transpilePackages
